system/system_test.go

```diff
@@ -0,0 +1,51 @@
+package system
+
+import (
+	"reflect"
+	"runtime"
+	"testing"
+)
+
+type noInputs func() string
+
+// test that a function with no inputs returns one of the expected strings
+func testOutputs(f noInputs, validOutputs []string, t *testing.T) {
+	output := f()
+	// use reflect to get the name of the function
+	name := runtime.FuncForPC(reflect.ValueOf(f).Pointer()).Name()
+	failed := true
+	for _, valid := range validOutputs {
+		if output == valid {
+			failed = false
+		}
+	}
+	if failed {
+		t.Errorf("Function %v returned %v, which is not one of %v", name, output, validOutputs)
+	}
+}
+
+func TestPackageManager(t *testing.T) {
+	t.Parallel()
+	testOutputs(DetectPackageManager, []string{"deb", "rpm", "apk", "pacman"}, t)
```

Comment:
Haven't run this yet, but it seems to be testing whether goss can detect the developer workstation. What happens if I run this test on a gentoo or OSX box?